src/rpp/rpp/operators/take.hpp

```diff
@@ -19,13 +19,22 @@ IMPLEMENTATION_FILE(take_tag);
 
 namespace rpp::details
 {
+
+template<constraint::decayed_type Type>
 class take_on_next
 {
 public:
-    take_on_next(size_t count) : m_count{ count } {}
+    explicit take_on_next(size_t count) : m_count{ count } {}
+
+    size_t emission_count() const
+    {
+        return m_count;
+    }
 
     void operator()(auto&& value, const constraint::subscriber auto& subscriber) const
     {
+        if (!subscriber.is_subscribed()) return;
```

Comment:
You don't need this check at all,

Reply:
Oh why? It's for the case that this is subscribed to an unsubscribed subscriber.